apps/dashboard/src/app/(dashboard)/event/components/qr-code-scanner.tsx

```diff
@@ -1,11 +1,19 @@
-import { Button } from "@mantine/core"
-import { type FC, useState } from "react"
+import { AspectRatio, Button, Collapse, Group, Loader, Skeleton, Stack, Text } from "@mantine/core"
+import { useDisclosure } from "@mantine/hooks"
+import { type FC, useEffect, useState } from "react"
 import { useZxing } from "react-zxing"
 import { useUpdateEventAttendanceMutation } from "../mutations"
 
 export const QrCodeScanner: FC = () => {
   const registerAttendance = useUpdateEventAttendanceMutation()
-  const [scannerOpen, setScannerOpen] = useState(false)
+  const [scannerOpen, { toggle }] = useDisclosure(false)
+  const [videoReady, setVideoReady] = useState(false)
+
+  useEffect(() => {
+    if (!scannerOpen) {
+      setVideoReady(false)
+    }
+  }, [scannerOpen])
```

Comment:
State updating state is an antipattern. It's better to just have the caller of setScannerOpen also call setVideoReady in this case.